### ISSUE TITLE
Problem with initial number key of object

### ISSUE THREAD

**Issue by giper45** [rhalff/dot-object#24]
Let's suppose the following dot object : 
```
args.1_telnet_server="server" 
args.2_name="username"
args.3_password="password"
args.4_commands="commands" 
```
What is the expected result ? 
Javascript gives an error when I try to set an object like this : 

```
var obj = {'1_ciao' : "ciao"} 
console.log(obj[1_ciao])

```
Instead the dot-object library returns this : 
```
  var args = dot.object(dottedArgs)
=>  
[ '1_telnet_server': 'server',
  '2_name': 'username',
  '3_password': 'password',
  '4_commands': 'commands' ]
```
A kind of array but that is not iterable. Anyway, I should order a set of keys like this in some way. If I try to use the keys without number (telnet_server, name , ...)  the application returns args dot in this order: [commands, name, password, telnet_server] ... I expect the order that I gave with the number convention that I cannot use with JS.  Do you have some  advice ? 
Thanks in advance

**Comment by aakrem** [comment]
Same for objects having numbers as keys will be transformed to an array 
```JavaScript
const foo = {"4": 20}
const dotObject = dot.object(foo)
// [ , , , , 20 ] array with 20 in the 4th item
```

**Comment by rhalff** [comment]
Fixed by #31  thanks for reporting.
